math/coin_problem.cpp

```diff
@@ -0,0 +1,83 @@
+/**
+ * @file
+ * @brief A program to solve the coin problem (https://en.wikipedia.org/wiki/Coin_problem)
+ * @details
+ * The coin problem is a math riddle also known as the Frobenius coin problem,
+ * postage stamp problem, and Chicken McNugget Theorem. The problem asks that
+ * given 2 items with different positive weights/value (for example coins/stamps with monetary value),
+ * what is the largest total weight/value that cannot be achieved using any combination of
+ * those two items? If the two items (a & b) are relatively prime, this can be solved using the
+ * formula (a*b)-(a+b). If they are not relatively prime, there is no solution.
+ * @author [Jason Baron](https://github.com/pkyu)
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for std::cin and std::cout
+
+ /**
+  * @namespace math
+  * @brief Mathematical algorithms
+  */
+namespace math {
+
+    /**
+     * @brief Determine if the numbers are relatively prime.
+     * If they are, return the solution using Frobenius number formula.
+     * If not, there is no solution. Return -1.
+     * @param a,b input numbers
+     * @return answer
+     */
+
+    int coin_problem(int a, int b) {
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). **Let us know if you need any help here or in our [Discord server](https://discord.gg/c7MnfGFGa6).** 🙂